warehouse/oidc/models.py

```diff
@@ -237,3 +258,35 @@ def job_workflow_ref(self):
 
     def __str__(self):
         return f"{self.workflow_filename} @ {self.repository}"
+
+
+class GitHubProvider(GitHubProviderMixin, OIDCProvider):
+    __tablename__ = "github_oidc_providers"
+    __mapper_args__ = {"polymorphic_identity": "github_oidc_providers"}
+    __table_args__ = (
+        UniqueConstraint(
+            "repository_name",
+            "repository_owner",
+            "workflow_filename",
+            name="_github_oidc_provider_uc",
+        ),
+    )
```

Comment:
Flagging for review: I had to dupe this `UniqueConstraint` between `GitHubProvider` and `PendingGitHubProvider`, even though they're functionally equivalent, thanks to the unique `name` requirement. There might be a better way to do this, though.

Reply:
I think this is due to not having set any naming_conventions for this project in the global metadata setup https://github.com/pypi/warehouse/blob/d58e6b241ce6fd2d65da469c32189b51a5572ff8/warehouse/db.py#L77
Alembic docs go into it here: https://alembic.sqlalchemy.org/en/latest/naming.html

I think we always named them manually, like here: https://github.com/pypi/warehouse/blob/d58e6b241ce6fd2d65da469c32189b51a5572ff8/warehouse/accounts/models.py#L197

So it might be a future refactoring to replace the manually-named constraints, but I’d approach that with some serious caution 😉